fix(provisioning_api): Don't allow to configure the same additional e…

nickvergessen opened the pull request:
…mail multiple times

Before | After
---|---
![Bildschirmfoto vom 2023-03-24 12-36-46](https://user-images.githubusercontent.com/213943/227511414-36b5c2a4-9219-4a01-956e-d1fec54a4243.png) | ![Bildschirmfoto vom 2023-03-24 12-35-46](https://user-images.githubusercontent.com/213943/227511425-805ae880-4c52-42db-a778-39c61a4d2229.png)

Editing one of the duplicated entries removed duplicates

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [x] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- [x] Screenshots before/after for front-end changes
- [x] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required
- [x] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)

nickvergessen committed:
fix(provisioning_api): Don't allow to configure the same additional email multiple times

Signed-off-by: Joas Schilling <coding@schilljs.com>

max-nextcloud commented:
I wonder if one could alter the email address in ways that create a different string but still result in the same email address:
me+some-mail-servers-ignore-this@nextcloud.com
or Me@nextcloud.com and me@Nextcloud.com

nickvergessen commented:
Can not reproduce the CI failure locally, so rebased to see if that solves it

nickvergessen commented:
Okay, tests are not isolated in server. executing the full file makes it fail, so fixing now

nickvergessen committed:
fix(tests): Fix new integration tests

The integration tests are not isolated, so we can not reuse previously used data

Signed-off-by: Joas Schilling <coding@schilljs.com>
